pulp_container/app/tasks/sync_stages.py

```diff
@@ -129,10 +130,13 @@ async def run(self):
                 tag_dc = DeclarativeContent(Tag(name=tag_name))
 
                 content_data = json.loads(raw_data)
-                media_type = content_data.get("mediaType")
+                media_type = content_data.get("mediaType") or dl_res.headers.get("content-type")
+                if media_type is None:
```

Comment:
I don't think you can raise an error here. It is totally valid for an OCI manifest not to have specified mediaType and for the content-type header - it is not a non required headers, so it can also miss. In that case, as the last resort thing, the recipient( us) can guess the media type by inspecting the contents of the body.
I am suggesting not doing this now, since most of the server usually always send content-type in the response.
You can instead strengthen the if condition like this:
`if media_type in (MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI) or 'manifests' in content_data.keys():`